### PR TITLE
Cli

### PR DESCRIPTION
**Pull request by sigvef**
Depends on #196 and #194.

image

image

### CODE REVIEW

**Comment by stianjensen** [comment]
How about using chalk for colored terminal output. I find it less intimidating than ` (\033[` and the cursor's with numbers.

https://www.npmjs.com/package/chalk

**Comment by sigvef** [comment]
ping, I changed from colors to chalk (it won't affect the cursor positioning code though).

**Comment by sigvef** [comment]
The prq dependencies have been resolved too, so this is ready for review.

File: nin/backend/serve.js [hunk 1]
```diff
@@ -123,9 +122,9 @@ var serve = function(projectPath, shouldRunHeadlessly) {
     files.listen(9000);
 
     if(shouldRunHeadlessly) {
-      console.log('running nin headlessly');
+      console.log('Running nin headlessly');
     } else {
-      console.log('serving nin on http://localhost:8000');
+      console.log('Serving nin on http://localhost:8000'.yellow);
```
Comment (stianjensen) [inline nin/backend/serve.js]: This outputs `undefined` for me.

**Comment by sigvef** [comment]
ping

File: nin/backend/utils.js [hunk 2]
```diff
@@ -8,14 +9,14 @@ function findProjectRootOrExit(currentPath) {
...
-      console.log('Found nin.json in project root (', base, ')');
+      console.log(chalk.grey('Found nin.json in project root (' + base + ')'));
       return base;
     }
 
     up += '../';
   } while (base != path.sep)
 
-  console.error('Could not find project root containing nin.json (looked from ', currentPath, ')');
+  console.error(('Could not find project root containing nin.json (looked from ' + currentPath + ')').red);
```
Comment (stianjensen) [inline nin/backend/utils.js]: Also prints undefined

File: nin/backend/serve.js [hunk 3]
```diff
@@ -123,9 +122,9 @@ var serve = function(projectPath, shouldRunHeadlessly) {
     files.listen(9000);
 
     if(shouldRunHeadlessly) {
-      console.log('running nin headlessly');
+      console.log('Running nin headlessly');
     } else {
-      console.log('serving nin on http://localhost:8000');
+      console.log(chalk.yellow('Serving nin on http://localhost:8000'));
```
Comment (stianjensen) [inline nin/backend/serve.js]: Getting `Chalk is not defined` here.

File: nin/backend/socket.js [hunk 4]
```diff
@@ -30,7 +30,7 @@ function socket(projectPath, onConnectionCallback) {
 
     conn.on('close', function () {
       delete connections[conn.id];
-      console.log('lost connection');
+      console.log('A client has disconnected.'.magenta);
```
Comment (stianjensen) [inline nin/backend/socket.js]: Getting `undefined` here

File: nin/backend/socket.js [hunk 5]
```diff
@@ -15,7 +15,7 @@ function socket(projectPath, onConnectionCallback) {
 
   server.on('connection', function (conn) {
     connections[conn.id] = conn;
-    console.log('connection!');
+    console.log('A client has connected.'.magenta);
```
Comment (stianjensen) [inline nin/backend/socket.js]: Getting `undefined` here

File: nin/backend/watch.js [hunk 6]
```diff
@@ -37,7 +36,7 @@ function watch(projectPath, cb) {
     }
 
     if (logFileChanges) {
-      console.log('Change in project detected: ' + event + ', ' + path);
+      console.log('Change in project detected: '.yellow + event.cyan + ', ' + path.magenta);
```
Comment (stianjensen) [inline nin/backend/watch.js]: This seems to log `NaN, undefined`

File: nin/backend/watch.js [hunk 7]
```diff
@@ -63,7 +62,7 @@ function watch(projectPath, cb) {
   shaderWatcher.on('all', function(event, path) {
     if (event === 'add' || event === 'change') {
       var pathParts = path.split('/');
-      console.log('Recompiling shaders:', pathParts[2]);
+      console.log('Recompiling shaders:'.yellow, pathParts[2].magenta);
```
Comment (stianjensen) [inline nin/backend/watch.js]: prints `undefined undefined`

**Comment by sigvef** [comment]
ping

**Comment by sigvef** [comment]
ping

**Comment by sigvef** [comment]
ping

**Comment by sigvef** [comment]
ping

**Comment by stianjensen** [comment]
conflicts :/

but otherwise it looks good :+1: :+1:

**Comment by sigvef** [comment]
ping

**Pull request merged**